dicesGame/src/test/java/com/example/dices/controllers/DiceControllerIntegrationTest.java

```diff
@@ -0,0 +1,266 @@
+package com.example.dices.controllers;
+
+
+import com.example.dices.models.DiceModel;
+import com.example.dices.models.JsonModel;
+import com.example.dices.repositories.IDiceRepository;
+import com.example.dices.services.DiceService;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.MvcResult;
+import org.springframework.test.web.servlet.request.MockMvcRequestBuilders;
+import org.springframework.test.web.servlet.result.MockMvcResultMatchers;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+
+@SpringBootTest
+@AutoConfigureMockMvc
+class DiceControllerIntegrationTest {
+    @Autowired
+    private MockMvc mockMvc;
+    @Autowired
+    private IDiceRepository diceRepository;
+
+    @Autowired
+    private DiceService diceService;
+
+    @Autowired
+    private ObjectMapper objectMapper;
+
+    public static class TestConstants {
+        public static final int DEFAULT_DICE_ID1 = 1;
+        public static final int DEFAULT_DICE_SIZE1 = 6;
+        public static final int DEFAULT_DICE_ID2 = 5;
+        public static final int DEFAULT_DICE_SIZE2 = 9;
+        public static final int DEFAULT_DICE_NOT_EXIST = 5;
+        public static final int DEFAULT_DICE_INVALID_SIZE = 1001;
+        public static final String ERROR_MESSAGE_1 = "There is no information to display";
+        public static final String ERROR_MESSAGE_2 = "Dice not found with ID: ";
+        public static final String ERROR_MESSAGE_3 = "The dice size must be between 1 and 1000.";
+
+    }
+
+    @BeforeEach
+    public void setUp() {
+        // Initialize Databases
+
+        DiceModel diceModel = new DiceModel();
+        diceModel.setDiceId(TestConstants.DEFAULT_DICE_ID1);
+        diceModel.setDiceSize(TestConstants.DEFAULT_DICE_SIZE1);
+        diceModel.setDiceId(TestConstants.DEFAULT_DICE_ID2);
+        diceModel.setDiceSize(TestConstants.DEFAULT_DICE_SIZE2);
+        diceRepository.save(diceModel);
+    }
+
+    @AfterEach
+    public void cleanDatabase() {
+        // Clean repository after each test.
+        diceRepository.deleteAll();
+    }
+
+
+    @Test
+    void testGetDicesNoEmpty() throws Exception {
```

Comment:
Please name the test cases this way: `UnitOfWork_StateUnderTest_ExpectedBehavior`.
For example: `testGetDices_diceExistsInDatabase_returnDice`.
See: https://stackoverflow.com/questions/155436/unit-test-naming-best-practices